src/main/java/me/xhawk87/CreateYourOwnMenus/Menu.java

```diff
@@ -253,6 +307,101 @@ private void onReload(FileConfiguration data) {
+     */
+    private void hasItemPlaceholders(int slot, ItemStack item){
+        //check if placeholders are enabled!
+        if (!Bukkit.getPluginManager().isPluginEnabled("PlaceholderAPI"))
+            return;
+
+
+        Placeholder placeholder = new Placeholder(slot, item);
+        boolean retVal = false;
+
+        //check if title has an placeholder in it
+        if(PlaceholderAPI.containsPlaceholders(item.getItemMeta().getDisplayName())) {
+            placeholder.title = item.getItemMeta().getDisplayName();
+            retVal = true;
+        }
+
+        //check if placeholders can be found in a lore
+        int index = 0;
+        if(item.getItemMeta().getLore() != null)
+            for(String s : item.getItemMeta().getLore()){
+                if(PlaceholderAPI.containsPlaceholders(s)){
+                    placeholder.lorePlaceholderPositions.put(index,s);
+                    retVal = true;
+
+                }
+                index++;
+            }
+
+        //if we have found a placeholder lets add it
+        if(retVal){
+            placeholderList.add(placeholder);
+        }
+
+    }
+
+    /**
+     * Placeholder Data and simple inner class
+     */
+
+
+    private List<Placeholder> placeholderList = new ArrayList<Placeholder>();
+
+    private class Placeholder{
+        private Placeholder( int slot, ItemStack item){
+            this.slot = slot;
+            this.item = item;
+        }
+        private ItemStack item; //backing up the item
+        private int slot;
+        private String title;
+        private Map<Integer,String> lorePlaceholderPositions = new TreeMap<>();
+        //private List<Integer> lorePlaceholderPositions = new ArrayList<>();
+
+
+    }
+
+
+    /**
+     * This function is called when a placeholder is used, therefore it changes every time the menu is opened
+     */
+    private void reArrangePlaceholders(final Player player){
+        if(placeholderList.isEmpty()) return;
+
+        //when Player is editing, display the normal lore and title, therefore %asdf% is displayed
+        if(isEditing(player)){
+           restoreInvPlaceholders();
+           return;
+        }
+
+        for(Placeholder p : placeholderList){
+            //create a copy of our item, so we don't override something important
+            ItemStack item = p.item;
+            ItemMeta meta = item.getItemMeta();
+
+            if(p.title != null){
+                String title = PlaceholderAPI.setPlaceholders(player,p.title);
+                meta.setDisplayName(title);
+            }
+
+            List<String> metaLore = meta.getLore();
+            if(!p.lorePlaceholderPositions.isEmpty()){
+                for(Map.Entry<Integer, String> entry : p.lorePlaceholderPositions.entrySet()){
+                    Integer slot = entry.getKey();
+                    String lore = entry.getValue();
+                    metaLore.set(slot,PlaceholderAPI.setPlaceholders(player,lore));
```

Comment:
```suggestion
                    metaLore.set(slot,PlaceholderAPI.setPlaceholders(player, lore));
```

Auto-formatting the page would probably help resolve a lot of these minor issues with newlines and whitespaces. Just to keep things neater for the next people to work on it.